### DISCUSSION TITLE
Scroll to a specific text in the document & highlight it?

### DISCUSSION THREAD

**Discussion by aquibbaig** [ueberdosis/tiptap#4768] [General Questions]
Say I have some text which I would like to scroll to in the editor when the page loads (and possibly highlight it), is there any API or way for implementing this ?

**Comment by abecirovic3** [comment]
I haven't found anything out of the box for this but since I have a pretty similar behavior in my project I will try to explain how I did it. In my case, we have a certain number of blocks that should be scrolled into view and highlighted when another block is clicked. The way I implemented this is I added an attribute to all of those types of blocks which is a unique ID, and then when the trigger block is clicked I just query the editor element for that particular element for example
`const el = editor.querySelector('[my-id="12345"]');`
and then scroll it into view with
```
el.scrollIntoView()
// attach class for highlight animation
```
Of course, I need to persist the information about the target block ID in the trigger block, so for the trigger block, I also implemented a custom extension that has an attribute for storing the information about the ID.

So I guess you can somehow mark the text that needs to be scrolled into view and then just use JS.
